### ISSUE TITLE
Forms: Dynamic forms and groups

### ISSUE THREAD

**Issue by arkadiuszwojcik** [OrchardCMS/OrchardCore#4250]
Right now when creating dynamic forms by Form widget and `FlowPart` it is impossible to make form groups. By group I mean common scenario when for example `Label` `Input` `Validation` elements are group be one html element.  Question is how such scenario schould be achived? Maybe there should be another widget `FormGroup` acceppting form fields widgets and while rendering it creates just one html outer element and renders insidel all childs?

**Comment by jptissot** [comment]
You could create your own widget for this

**Comment by arkadiuszwojcik** [comment]
Yes I could and for now I will probably do it, problem is that current OrchardCore.Forms module implemantations have some problems with most basic forms setup. What I would suggest is:

1. In `Form-FlowPart.cshtml` remove part that renders this line: `widgetContent.Classes.Add("form-group");`

2. Create `FormGroup ` widged with `FlowPart`/`BagPart ` to render `<div class="form-group"..> ... </div>`

3. Would be nice if flowMetadata could contain option to skip rendering widget outer div: `<div class="widget ..." ..> ... </div>`.  So when I have `Label ` `Input ` and `Validation` I could render them in `FormGroup ` one by one without wrapping each of them in extra widget div

**Comment by jptissot** [comment]
I agree with you. I had to write most of my widgets for a form.

**Comment by arkadiuszwojcik** [comment]
I investigated few forms modules from different CMS frameworks. One thing worth to note is that all modules I inspected never separate label from actual input field. So basically label is integral part of form element with option to render it or not and also placement option: before or after input field. In most cases it was also true for validation message field. I think such solution makes more sense then current design. Right now when someone would like to render Label and Input both of them will be wrapped in few levels of mambo jambo divs.

**Comment by jptissot** [comment]
Yes, This is eactly what I am doing with my custom widgets. I have a bunch of options and Label / Input are rendered within the same input. I also have support for HTML5 validation attributes.

**Comment by Piedone** [comment]
Have you tried the built-in basic Container widget? Might not be everything you need but it still provides basic grouping capabilities.

**Comment by arkadiuszwojcik** [comment]
@Piedone my concern is about first time use of Forms module and some default settings for new users. After few days I was able to do things my way but I had to dig in into source code. For example I had to learn that I need to override `Widget_Label`, `Widget_Input `... to skip default widget wrapper around each form element. Also probably Container widget might work but it is not avaliable when installing form module, I don't even know which module Container widget is in. To sum up this thread, I would like to give my thougs here about form module usage after my first experience:

1. Override default forms elements widgets (Widget_Label, Widget_Input ...) to skip wrappers rendering.

2. Add Container/FormGroup? widget to Forms module, and maybe Fieldset as well.

3. Consider not spliting Input/TextArea field and Labels as separate widgets. Use Input fields with option to render Label: a) Above input b) Under input c) Not render
